Makefile

```diff
@@ -250,22 +266,37 @@ depend.update.full: depend.cleanlock depend.update
 depend.cleanlock:
 	-rm Gopkg.lock
 
-depend.update:
+depend.update: | $(DEP)
 	@echo "Running dep ensure with DEPARGS=$(DEPARGS)"
-	time dep ensure $(DEPARGS)
+	time $(DEP) ensure $(DEPARGS)
 	cp Gopkg.* vendor/
 	@echo "now check the diff in vendor/ and make a PR"
 
-vendor.check:
-	@echo "Checking that Gopkg.* are in sync with vendor/ submodule:"
-	@echo "if this fails, 'make pull' and/or seek on-call help"
-	diff Gopkg.toml vendor/
-	diff Gopkg.lock vendor/
-
 .PHONY: vendor.check
+vendor.check: $(ISTIO_OUT)/vendor_checked
+
+$(ISTIO_OUT)/vendor_checked: Gopkg.toml Gopkg.lock | $(ISTIO_OUT) $(GRPC_DIR)
+	@diff Gopkg.toml vendor/ || (echo "vendor/ out-of-sync, run 'make pull' and/or seek on-call help" && exit 1)
+	@diff Gopkg.lock vendor/ || (echo "vendor/ out-of-sync, run 'make pull' and/or seek on-call help" && exit 2)
+	touch $(ISTIO_OUT)/vendor_checked
 
 ${GEN_CERT}:
-	unset GOOS && unset GOARCH && CGO_ENABLED=1 bin/gobuild.sh $@ istio.io/istio/pkg/version ./security/cmd/generate_cert
+	GOOS=$(LOCAL_GOOS) GOARCH=$(LOCAL_GOARCH) bin/gobuild.sh $@ istio.io/istio/pkg/version ./security/cmd/generate_cert
+
+# If CGO_ENABLED=0 then go get tries to install in system directories.
+# If -pkgdir <dir> is also used then various additional .a files are present.
+${ISTIO_BIN}/dep:
+	unset GOOS && unset GOARCH && CGO_ENABLED=1 go get -u github.com/golang/dep/cmd/dep
+
+fortio ${ISTIO_BIN}/fortio: | $(ISTIO_OUT)/vendor_checked ${ISTIO_BIN}
+	unset GOOS && unset GOARCH && cd vendor/istio.io/fortio/ && go install .
+
+${ISTIO_BIN}/helm$(HELM_VER): | ${ISTIO_BIN} $(FORTIO)
+	cd /tmp && \
+        $(FORTIO) curl -L -stdclient -timeout 15m https://storage.googleapis.com/kubernetes-helm/helm-${HELM_VER}-${LOCAL_GOOS}-amd64.tar.gz > /tmp/helm.tgz && \
```

Comment:
why 15 minutes ? if we don't read anything for 15 minutes there is a network/connection problem...
make a new variable
CURL:=$(FORTIO) curl -L -timeout 1min
that people are free to swap for CURL=curl

why not | tar directly ?

Reply:
At home on wifi some of these downloads were taking on the order of 10 min, so I figured 15 min might be needed for anyone trying to work on istio in a coffee shop.  Perhaps I misinterpreted the meaning of the timeout--I figured it was a cap on the total download time but if it's really measuring the time since last packet then I can crank down the value and verify it still works from a remote location.

I could probably pipe to tar.  I think I had mostly tried to preserve the existing implementation.

Reply:
(assuming we do any of this) let's do the variable and the pipe

or we can use a docker image for curl maybe

thanks I didn't actually test the timeout but you are right the std client timeout appears to be a total time - filled https://github.com/istio/fortio/issues/221